source/features/clean-image-auto-links.tsx

```diff
@@ -0,0 +1,32 @@
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {isSingleMarkdownFile} from '../github-helpers';
+import GitHubURL from '../github-helpers/github-url';
+
+function init(): void {
+	for (const image of select.all('.markdown-body a > img')) {
+		const link = image.parentElement as HTMLAnchorElement;
+
+		if (image.src === link.href) {
+			link.replaceWith(image);
+		}
+
+		if (pageDetect.isSingleFile(link)) {
+			const rawUrl = new GitHubURL(link.href).assign({route: 'raw'});
+
+			if (image.src === rawUrl.href) {
+				link.replaceWith(image);
+			}
+		}
+	}
+}
+
+void features.add(__filebasename, {
+	include: [
+		pageDetect.isRepoTree,
+		() => pageDetect.isSingleFile() && isSingleMarkdownFile()
```

Comment:
isSingleMarkdownFile should be enough. It already implies that it’s a single file.

Reply:
There are both `isSingleFile` and `isEditingFile` combined with `isSingleMarkdownFile`, so I didn't put `isSingleFile` into `isSingleMarkdownFile`. But perhaps it should be called `isMarkdownFile` from the beginning 😅